### PR TITLE
Implement Custom Permission Popover for Camera and Audio Access

### LINKED ISSUE #21: Feature Request: Custom Popover for Camera and Microphone Access
**Is your feature request related to a problem? Please describe.**

To enhance user experience and ensure privacy transparency, I would like to implement a custom popover that explicitly asks for user permission to access the camera and microphone. The popover should provide clear context as to why the app needs these permissions, giving users full control over their privacy settings before initiating any media capture.

**Describe the solution you'd like**

A clean, easy-to-understand popover that prompts users to grant access to their camera and microphone. The popover will have clear messaging, either camera or audio access is grannted or denied, providing users control over their privacy.

**Describe alternatives you've considered**

Using the default browser permission request alone could work, but it lacks the customization and contextual information that helps users understand why access is being requested. (Current implementation)

** I have already tested out the component , and you can assign it to me under the hacktoberfest tag
** I have a attached a prototype of the component for better understanding

Screenshot (1251)

### PR DESCRIPTION
**Pull request by Suman-Jyoti-Phukan**
## Description

This update introduces a custom permission popover to request user permission for camera and audio access. The component is designed to streamline the user experience by managing permissions within the active session and minimizing redundant prompts. Initially , I thought it would take me a couple hours but I was wrong. It became more complicated but at the end it worked out fine. And sorry , I had to PR with my master branch as I was too focus on developing it.

Fixes #21 

## Changes

List the changes done to fix a bug or introducing a new feature.

I had to apply many chances and introduce new states,effects that hadles various situations. There were few challenges while creating this component. 

- Introduced multiple components to manage and display the user’s current permission status, along with prompts to request permissions as needed.
- Ensured cross-browser compatibility for consistent behavior across different environments.
- Implemented multiple useEffect hooks and state variables to handle various permission and event states reliably.
- The if conditions are a bit to hard to explain without actually running the program. So, I have left multiple comments. 

## How to test

Simply open the development server and hit the join button at https://localhost:3000/chat, you will notice a custom permission component.

## Checklist:

- [Y] My code follows the style guidelines of this project
- [Y] I have performed a self-review of my code
- [Y] I have commented my code, particularly in hard-to-understand areas
- [N] I have made corresponding changes to the documentation
- [Y] My changes generate no new warnings
- [Y] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Comment by kaali001** [comment]
@Suman-Jyoti-Phukan Its nice PR,   But its need few modifications, make changes as given below -

1. Make a new folder named as `media_permission` for your new files created in component folder.

2. Use  this in the UI of pop up card - 
    - color pallet - #d53b84,#fa546b, #c14020, #f1633d
    - Fonts - Cinzel, Quattrocento, Ultine Extended Demi

3. Fix the issue - on clicking the chat button the chat getting started even if the popup for permission is there, Actually by clicking on chat button it directly makes chat active and call ` room` component even without permission. You can solve this by adding a step in between that if the all permission is allowed then it makes chat active `true` .  

4. Instead of adding many functions on the `chat.tsx` file you can create a new file in `media_permission` for handling all this and keep  only components call on `chat.tsx` file ,  as it's a inappropriate that you crated 4 new files and still defining many function in `chat.tsx` .

**Comment by Suman-Jyoti-Phukan** [comment]
Hey , thanks for the update. I'll make the appropriate changes you asked. And for the point 3, it was already directly making the chat active and call  room component even without permission. I had noticed it but decided not to fix it. However , It can be fixed with few changes.

**Comment by Suman-Jyoti-Phukan** [comment]
Hey can clarify on thing. You want a media_permission inside the components folder or in the src folder?

**Comment by Suman-Jyoti-Phukan** [comment]
Hey , I have made the changes you have asked. Please check it out and leave me a review. Thank you.

File: client/src/pages/Room.tsx
Comment (kaali001) [inline client/src/pages/Room.tsx:1]: @Suman-Jyoti-Phukan , This file doesn't require any changes, Please let me know if you faced any issue with in this file, as I go through this file it seems you made changes in the text inline, I will remove this file from PR. Waiting for your response.

**Review by kaali001 — CHANGES_REQUESTED** [review]
@Suman-Jyoti-Phukan , This PR resolved all existing issue, just remove the `Room.tsx` file  from PR as it does not need to be change , And change the file type of  `mediaPermissions.ts` file to `.tsx` . Then its ready to be merge.

**Comment by Suman-Jyoti-Phukan** [comment]
I have done the changes. Can you check it now?

**Comment by kaali001** [comment]
@Suman-Jyoti-Phukan , You removed all the codes of  `Room.tsx` file ,  that's why Vercel — Deployment has failed. 
You don't need to remove all codes from file you just have to use original code of this file. you can get original code from my master branch.

**Comment by Suman-Jyoti-Phukan** [comment]
Never mind. I thought you told me to delete the file. I will update it and re commit it.

**Comment by Suman-Jyoti-Phukan** [comment]
Hey @kaali001 , I have made the changes. Do have a look at it. Thank you.

**Review by kaali001 — APPROVED** [review]

**Comment by kaali001** [comment]
@Suman-Jyoti-Phukan , It was a nice contribution. I have merged the PR by making the `room.tsx` file as it was still not updated by you.    Now  can you  open a new issue for fixing the ui of the permission popup card as it looks kind of ugly.

**Comment by Suman-Jyoti-Phukan** [comment]
Thanks. It was great collaborating with you. I will work on it as soon as possible. However, it would be great if you can provide some insight on how to style it.

**Pull request merged**